### ISSUE TITLE
dQ data is not handled properly when loaded from CanSAS1D and NXcanSAS (Trac #826)

### ISSUE THREAD

**Issue by smk78** [SasView/sasview#921]
The ISIS SANS interface in Mantid now permits the calculation of Qres (dQ) data which can then be written to CanSAS1D XML or NXcanSAS. In testing compatibility with !SasView I have observed the following:

Loading from CanSAS1D:

!SasView 4.0.1 - works fine! On selecting a model and checking Use dQ Data the dQ data range is displayed. Fitting proceeds ok.

!SasView 4.1 local build - On opening a file containing dQ data !SasView throws the error:

   2016-12-15 15:48:45 : The following errors occured whilst loading
   41951_1D_qres.xml:
   Error: 
   The loaded xml file, 41951_1D_qres.xml does not fully meet the CanSAS
   v1.x specification. !SasView loaded as much of the data as possible.
   Element '{urn:cansas1d:1.1}name': This element is not expected.
   Expected is one of ( {urn:cansas1d:1.1}length,
   {urn:cansas1d:1.1}aperture )., line 136

The data is still read and can be sent to fitting but the dQ data is not available.

Loading from NXcanSAS:

!SasView 4.0.1 - The data is read and can be sent to fitting but the dQ data is not available to !SasView.

!SasView 4.1 local build - The data is read and can be sent to fitting but the dQ data is not available to !SasView.

Migrated from http://trac.sasview.org/ticket/826
```json
{
    "status": "closed",
    "changetime": "2017-01-23T18:26:08",
    "_ts": "2017-01-23 18:26:08.794400+00:00",
    "description": "The ISIS SANS interface in Mantid now permits the calculation of Qres (dQ) data which can then be written to CanSAS1D XML or NXcanSAS. In testing compatibility with !SasView I have observed the following:\n\n\nLoading from CanSAS1D:\n\n!SasView 4.0.1 - works fine! On selecting a model and checking Use dQ Data the dQ data range is displayed. Fitting proceeds ok.\n\n!SasView 4.1 local build - On opening a file containing dQ data !SasView throws the error:\n\n   2016-12-15 15:48:45 : The following errors occured whilst loading\n   41951_1D_qres.xml:\n   Error: \n   The loaded xml file, 41951_1D_qres.xml does not fully meet the CanSAS\n   v1.x specification. !SasView loaded as much of the data as possible.\n   Element '{urn:cansas1d:1.1}name': This element is not expected.\n   Expected is one of ( {urn:cansas1d:1.1}length,\n   {urn:cansas1d:1.1}aperture )., line 136\n\nThe data is still read and can be sent to fitting but the dQ data is not available.\n\n\nLoading from NXcanSAS:\n\n!SasView 4.0.1 - The data is read and can be sent to fitting but the dQ data is not available to !SasView.\n\n!SasView 4.1 local build - The data is read and can be sent to fitting but the dQ data is not available to !SasView.\n",
    "reporter": "smk78",
    "cc": "",
    "resolution": "fixed",
    "workpackage": "SasView Bug Fixing",
    "time": "2016-12-15T15:57:16",
    "component": "SasView",
    "summary": "dQ data is not handled properly when loaded from CanSAS1D and NXcanSAS",
    "priority": "blocker",
    "keywords": "",
    "milestone": "SasView 4.1.0",
    "owner": "krzywon",
    "type": "defect"
}
```

**Comment by smk78** [comment]
Trac update at `2016/12/15 15:57:58`: 
* **smk78** changed attachment from "" to "41951_1D_noqres.xml"
* **smk78** commented:

CanSAS1D without dQ values

**Comment by smk78** [comment]
Trac update at `2016/12/15 15:58:18`: 
* **smk78** changed attachment from "" to "41951_1D_qres.xml"
* **smk78** commented:

CanSAS1D with dQ values

**Comment by smk78** [comment]
Trac update at `2016/12/15 15:58:40`: 
* **smk78** changed attachment from "" to "41951_1D_noqres.h5"
* **smk78** commented:

NXcanSAS without dQ values

**Comment by smk78** [comment]
Trac update at `2016/12/15 15:59:04`: 
* **smk78** changed attachment from "" to "41951_1D_qres.h5"
* **smk78** commented:

NXcanSAS with dQ values

**Comment by smk78** [comment]
Trac update at `2016/12/15 15:59:46`: **smk78** changed summary from "dQ data is not handled properly when loaded from NXcanSAS" to "dQ data is not handled properly when loaded from CanSAS1D and NXcanSAS"

**Comment by smk78** [comment]
Trac update at `2016/12/20 12:36:57`: **smk78** commented:

Have just pulled and built !SasView locally, and downloaded Jenkins build 621. Both versions LOAD the XML and H5 datasets but DO NOT appear to use the dQ data - although the 'Use dQ' radio button is checked, no dQ values appear in the !FitPage.

**Comment by smk78** [comment]
Trac update at `2017/01/13 15:32:10`: **smk78** commented:

Pulled and built on Win7.

XML & H5 datasets load and can be fitted. Use dQ is automatically selected and fitting with it enabled does generate a slightly better fit than if it is not enabled.

However, loading the XML still throws the error in the original ticket. But you can OK and continue.

**Comment by smk78** [comment]
Trac update at `2017/01/21 12:59:21`: **smk78** commented:

Mantid team will endeavour to fix the CanSAS1D XML issue for v3.9.0 in Feb.
